src/main/java/org/snomed/snowstorm/core/data/services/MultiSearchService.java

```diff
@@ -75,23 +63,43 @@ public class MultiSearchService implements CommitListener {
 	
 	private final Logger logger = LoggerFactory.getLogger(getClass());
 	
-	Map<String, String> publishedBranches = new HashMap<>();
-	
-	BoolQueryBuilder cachedBranchesQuery = null;
-	LocalDate cacheDate = null;
+	private final Map<String, String> publishedBranches = new HashMap<>();
+
+	private MultiBranchCriteria cachedBranchCriteria = null;
+	private LocalDate cacheDate = null;
 
 	public Page<Description> findDescriptions(DescriptionCriteria criteria, PageRequest pageRequest) {
+
+		MultiBranchCriteria cachedBranchesQuery = getBranchesQuery();
 		
-		SearchHits<Description> searchHits = findDescriptionsHelper(criteria, pageRequest);
+		MultiBranchCriteria branchesQuery = new MultiBranchCriteria("all-released-plus-requested",cachedBranchesQuery.getTimepoint(),new ArrayList<>(cachedBranchesQuery.getBranchCriteria()));
+		if (criteria.getIncludeBranches() != null && criteria.getIncludeBranches().getBranches() != null) {
+			for(String branch : criteria.getIncludeBranches().getBranches()) {
+				if (!branchesQuery.getBranchCriteria().contains(versionControlHelper.getBranchCriteria(branch))) {
```

Comment:
Also calling `getBranchCriteria` has some cost. Better to assign the result to a variable and reuse it rather than calling this getter again on the next line.

Reply:
Contains() removed and replaced with check against branchPath, and getBranchCriteria saved to a variable.